io_scene_niftools/modules/nif_import/geometry/mesh/__init__.py

```diff
@@ -127,8 +129,7 @@ def import_mesh(self, n_block, b_obj):
                 normals = None
             else:
                 # for some reason, normals can be four-component structs instead of 3, discard the 4th.
-                if len(normals[0]) > 3:
-                    normals = [(n[0], n[1], n[2]) for n in normals]
+                normals = np.array(normals)[:, :3]
```

Comment:
if this was sent as an array to begin with, no need to cast

Reply:
It's not necessarily sent as an array (for example when gotten from a regular NiMesh, i.e. no displaylist, it's a list).
I do see that `nif_import/geometry/vertex/__init__.py` also does conversion to a numpy array, so this piece of code could potentially be moved down to apply to _all_ normals, such that the vertex import always gets a numpy array and doesn't have to do the conversion itself. The only downside would be that the indexing would happen on all different cases, even when it isn't necessary.

Reply:
Yeah, I was referring to the suggestion of sending all normals as np arrays regardless of original source.
I think this type of indexing only returns a view and not a copy, so it's super cheap. Not 100% sure though.

Reply:
I tried it out - at least with a simple test array, that did indeed return a view.